src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -77,92 +78,122 @@ public static IEnumerable<VisualStudioInstance> GetInstances(string workingDirec
             }
         }
 
+        enum hostfxr_resolve_sdk2_flags_t
+        {
+            disallow_prerelease = 0x1,
+        };
+
+        enum hostfxr_resolve_sdk2_result_key_t
+        {
+            resolved_sdk_dir = 0,
+            global_json_path = 1,
+        };
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        private delegate void hostfxr_resolve_sdk2_result_fn(
+                hostfxr_resolve_sdk2_result_key_t key,
+                string value);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        private delegate void hostfxr_get_available_sdks_result_fn(
+                hostfxr_resolve_sdk2_result_key_t key,
+                [MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 0)]
+                string[] value);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_resolve_sdk2(
+            string exe_dir,
+            string working_dir,
+            hostfxr_resolve_sdk2_flags_t flags,
+            hostfxr_resolve_sdk2_result_fn result);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_get_available_sdks(string exe_dir, hostfxr_get_available_sdks_result_fn result);
+
+        [DllImport("libc", ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr realpath(string path, IntPtr buffer);
+
+        [DllImport("libc", ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern void free(IntPtr ptr);
+
+        private static string realpath(string path)
+        {
+            IntPtr ptr = realpath(path, IntPtr.Zero);
+            string result = Marshal.PtrToStringAuto(ptr);
+            free(ptr);
+            return result;
+        }
```

Comment:
The .NET convention for this type of thing is to put these into a class named `NativeMethods`: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1060